c7n/resources/ec2.py

```diff
@@ -918,6 +918,43 @@ def process(self, instances):
                 Groups=groups[idx])
 
 
+@actions.register('associate-instance-profile')
+class EC2AssociateInstanceProfile(BaseAction):
+    """Associates an instance profile to an existing EC2 instance.
+
+    :Example:
+
+    .. code-block: yaml
+
+        policies:
+          - name:
+            resource: ec2
+            query:
+              - IamInstanceProfile: absent
+            actions:
+              - type: associate-instance-profile
+                name: default
+
+    https://docs.aws.amazon.com/cli/latest/reference/ec2/associate-iam-instance-profile.html
+    """
+
+    schema = type_schema(
+        'associate-instance-profile',
+        **{'name': {'type': 'string'}})
```

Comment:
wondering if we shouldn't also have this have the ability to yank a profile if None or empty string is provided. thoughts?